src/test/java/uk/gov/pay/ledger/pact/ContractTest.java

```diff
@@ -375,4 +394,30 @@ private void createPaymentTransactionForSelfserviceSearch(String transactionExte
                 .withCreatedDate(ZonedDateTime.parse(createdDate))
                 .insert(app.getJdbi());
     }
+
+    private void createPaymentTransactionWithAllFields(String transactionExternalId,
+                                                       String gatewayAccountId,
+                                                       TransactionState state,
+                                                       String reference,
+                                                       String createdDate) {
+        aTransactionFixture()
+                .withExternalId(transactionExternalId)
+                .withGatewayAccountId(gatewayAccountId)
+                .withTransactionType(TransactionType.PAYMENT.name())
+                .withAmount(1000L)
+                .withReference(reference)
+                .withState(state)
+                .withDefaultCardDetails()
+                .withFirstDigitsCardNumber("424242")
+                .withLastDigitsCardNumber("4242")
```

Comment:
Would it be possible to make the last digits not be a subset of the first digits? It could get confusing or allow bugs to sneak past if we accidentally grab from the wrong end of the card number.